### ISSUE TITLE
C/C++ Extension fails to download binaries on remote

### ISSUE THREAD

**Issue by RamyElkest** [microsoft/vscode-cpptools#5290]
Type: General

**Describe the bug**
- OS and Version: Windows 10 (Linux ssh remote)
- VS Code Version: 1.43
- C/C++ Extension Version: 0.27
- Other extensions you installed (and if the issue persists after disabling them): Remote SSH
- A clear and concise description of what the bug is.

**To Reproduce**
*Please include code sample and `task.json` files.*
1. Connect to Remote container with no internet access
2. Install C/C++ extension (with "remote.downloadExtensionsLocally": true)
3. Try to activate extension
4. Downloading the linux binaries should retry and fail

```
Updating C/C++ dependencies...

Downloading package 'C/C++ language components (Linux / x86_64)'  Failed. Retrying... Failed. Retrying... Failed. Retrying...Waiting 8 seconds... Failed. Retrying...Waiting 16 seconds...
```

**Expected behavior**
Extension should respect "remote.downloadExtensionsLocally": true and download locally then xfer

**Additional context**
Installing via offline vsix obviously works as a workaround. I'd just like to know what the long term plan is here.

**Comment by sean-mcmanus** [comment]
We're hoping the VS Code marketplace will enable OS-specific downloads so we can stop using our own downloading mechanism. @bobbrow Is there a public issue tracking that somewhere? I can't seem to find it.

**Comment by bobbrow** [comment]
I don't know of one.  We are tracking it in a private repository.

**Comment by sean-mcmanus** [comment]
@bobbrow Does the private repo have any ETA details yet?

**Comment by bobbrow** [comment]
No, there is no ETA yet.

**Comment by RamyElkest** [comment]
@sean-mcmanus is it possible to progress with this now that remote-connections declare the platform upon connection? (albeit not the full vscode solution i suppose)
`remote.SSH.remotePlatform`

**Comment by sean-mcmanus** [comment]
@RamyElkest It might be possible, but we were not planning to currently.

**Comment by aallrd** [comment]
Hello,
Coming from #6933 where I hit this issue.
My remote environment is offline and it consistently fails to install the `vscode-cpptools` extension when the others are okay.
Is there progress on the Marketplace feature to distribute OS-specific packages?

**Comment by sean-mcmanus** [comment]
@aallrd I think there was an issue on the VS Code or marketplace repo, but I can't seem to find it though.

**Comment by Colengms** [comment]
This should get addressed by: microsoft/vscode#23251

**Comment by bobbrow** [comment]
This will be addressed by #8152 starting in version 1.8.0
